dynamic-demo-plugin/README.md

```diff
@@ -65,6 +65,57 @@ spec:
 # ...
 ```
 
+## Proxy service
+
+In case the plugin needs to communicate with some in-cluster service, it can
+declare a service proxy in its `ConsolePlugin` resource using the
+`spec.proxy.services` array field. A service `name`, `namespace` and `port`
+needs to be specified.
+
+Console backend exposes following endpoint in order to proxy the communication
+between plugin and the service:
+`/api/proxy/namespace/<service-namespace>/service/<service-name>:<port-number>/<request-path>?<optional-query-parameters>`
+
+An example proxy request path from plugin to `helm-charts` service,
+in `helm` namespace to list ten helm releases:
+`/api/proxy/namespace/helm/service/helm-charts:8443/releases?limit=10`
+
+Proxied request will use [service CA bundle](https://docs.openshift.com/container-platform/4.8/security/certificate_types_descriptions/service-ca-certificates.html) by default. The service must use HTTPS.
+If the service uses a custom service CA, the `caCertificate` field
+must contain the certificate bundle. In case the service proxy request
+needs to contain logged-in user's OpenShift access token, the `authorize`
+field needs to be set to `true`. The user's OpenShift access token will be
+then passed in the HTTP `Authorization` request header, for example:
+
+`Authorization: Bearer sha256~kV46hPnEYhCWFnB85r5NrprAxggzgb6GOeLbgcKNsH0`
+
+```yaml
+# ...
+spec:
+  proxy:
+    services:
+    - name: helm-charts
+      namespace: helm
+      port: 8443
+      caCertificate: '-----BEGIN CERTIFICATE-----\nMIID....'
+      authorize: true
+# ...
+```
+
+### Local development
+
+In case of local developement of the dynamic plugin, just set up your
+HTTP server locally and pass its endpoint address in form of a service proxy 
+entry to the console server in form of JSON, using the `--plugin-proxy` flag.
+
+
+Example:
+```
+ ./bin/bridge --plugin-proxy={"services":[{"consoleAPIPath":"/api/proxy/namespace/serviceNamespace/service/serviceName:9991/","endpoint":"localhost:8080"}]}
```

Comment:
I'd think we'd need single quotes around the plugin proxy JSON here or the shell will split the argument.